app/Main.hs

```diff
@@ -1,9 +1,85 @@
 -- SPDX-FileCopyrightText: 2022 Tezos Commons
 -- SPDX-License-Identifier: LicenseRef-MIT-TC
 
+{-# LANGUAGE OverloadedLists #-}
+
 module Main
   ( main
   ) where
 
+import Lorentz (DGitRevision(..), defaultContract)
+import Lorentz.ContractRegistry
+import Main.Utf8 (withUtf8)
+import qualified Options.Applicative as Opt
+import Options.Applicative.Help.Pretty (Doc, linebreak)
+import System.Environment (withProgName)
+import Data.Char (isUpper, toLower)
+import GHC.TypeLits (symbolVal)
+
+import Morley.Util.Named
+import Morley.CLI
+
+import Indigo.Contracts.HomebaseLite
+
+programInfo :: DGitRevision -> Opt.ParserInfo CmdLnArgs
+programInfo gitRev = Opt.info (Opt.helper <*> argParser contracts gitRev) $
+  mconcat
+  [ Opt.fullDesc
+  , Opt.progDesc "This executable provides commands for development and interaction with the project."
+  , Opt.header "Homebase-Lite"
+  , Opt.footerDoc $ Just usageDoc
+  ]
+
+usageDoc :: Doc
+usageDoc = mconcat
+   [ "You can use help for specific COMMAND", linebreak
+   , "EXAMPLE:", linebreak
+   , "  homebase-lite print --help", linebreak
+   ]
+
+contracts :: ContractRegistry
+contracts = ContractRegistry $
+  [ "Homebase-Lite" ?:: ContractInfo
+    { ciContract = defaultContract homebaseLiteCode
+    , ciIsDocumented = True
+    , ciStorageParser = Just storageParser
+    , ciStorageNotes = Nothing
+    }
+  ]
+
+storageParser :: Opt.Parser Storage
+storageParser = initialStorage
+  <$> adminParser
+  <*> expireTimeParser
+  <*> voteDelayParser
+  <*> quorumThresholdParser
+  <*> minimumBalanceParser
+  where
+    adminParser = (#admin :!) <$> keyHashOption Nothing
+      (#name :! "admin")
+      (#help :! "Initial admin address")
+    toLong (Name :: Name s) = concat $ flip map (symbolVal $ Proxy @s) \case
+      x | isUpper x -> ['-', toLower x]
+        | otherwise -> [x]
+    naturalParser label deflt help =
+      fmap (label :!) .
+        Opt.option (Opt.maybeReader readMaybe) $
+          Opt.long (toLong label) <>
+          Opt.metavar "NATURAL" <>
+          Opt.value deflt <>
+          Opt.help help <>
+          Opt.showDefault
+    expireTimeParser = naturalParser #expireTime (7 * 24 * 3600)
+      "After how much time a proposal can no longer be voted on (in seconds)"
+    voteDelayParser = naturalParser #voteDelay (24 * 3600)
+      "After how much time votes can be cast on a proposal (in seconds)"
+    quorumThresholdParser = naturalParser #quorumThreshold 1
+      "The minimum amount of votes to be cast on a proposal for it to be successful, \
+      \expressed as a constant number to reach"
+    minimumBalanceParser = naturalParser #minimumBalance 1
+      "The minimum amount of governance tokens needed to submit a new proposal"
```

Comment:
:thinking: arguably `0` is a possible (ans sensible, if that's what one wants) value for all of these, so how about using that as the default?

Reply:
`quorumThreshold` of `0` doesn't really make sense AFAIU -- that would imply all proposals are automatically successful regardless of votes.

Reply:
well, yes, all proposals would be successful, but this wouldn't make them useless as there would still be the data of the choices made by those that participated (if any).

For example one may want to collect feedback from a community regardless of participation in the voting itself.

Arguably a bit of a stretch, but I think that "always successful proposals" are a valid option :shrug:

Reply:
Okay, fair.